starlette/responses.py

```diff
@@ -327,6 +327,22 @@ async def __call__(self, scope: Scope, receive: Receive, send: Send) -> None:
         )
         if self.send_header_only:
             await send({"type": "http.response.body", "body": b"", "more_body": False})
+        elif "http.response.zerocopysend" in scope["extensions"]:  # pragma: no cover
+            if os.name == "nt":
+                open_mode = os.O_RDONLY | os.O_BINARY
+            else:
+                open_mode = os.O_RDONLY
+            fd = await anyio.to_thread.run_sync(os.open, self.path, open_mode)
+            try:
+                await send(
+                    {
+                        "type": "http.response.zerocopysend",
+                        "file": fd,
+                        "more_body": False,
+                    }
+                )
+            finally:
+                await anyio.to_thread.run_sync(os.close, fd)
```

Comment:
I'm... not comfortable with this.
Not at all obvious to me how the open/zerocopesend/close would interact.

I've got a bit of a preference here that we close this off for now as a "feature pause" until we're sufficiently on top of all existing issues that we can *thoroughly* review this.